include/io/io_mesh_ascii.tcc

```diff
@@ -246,6 +246,94 @@ std::vector<Eigen::Matrix<double, 6, 1>>
   return stresses;
 }
 
+//! Return particles scalar properties
+template <unsigned Tdim>
+std::vector<std::tuple<mpm::Index, double>>
+    mpm::IOMeshAscii<Tdim>::read_particles_scalar_properties(
+        const std::string& scalar_file) {
+
+  // Particles scalar properties
+  std::vector<std::tuple<mpm::Index, double>> scalar_properties;
+  scalar_properties.clear();
+
+  // input file stream
+  std::fstream file;
+  file.open(scalar_file.c_str(), std::ios::in);
+
+  try {
+    if (file.is_open() && file.good()) {
+      // Line
+      std::string line;
+      while (std::getline(file, line)) {
+        boost::algorithm::trim(line);
+        std::istringstream istream(line);
+        // ignore comment lines (# or !) or blank lines
+        if ((line.find('#') == std::string::npos) &&
+            (line.find('!') == std::string::npos) && (line != "")) {
+          while (istream.good()) {
+            // ID
+            mpm::Index id;
```

Comment:
Could we move these out, otherwise we are creating them at every iteration of the loop?